### PR TITLE
[simdjson] Update to 2.2.0

### PR DESCRIPTION
**Pull request by myd7349**
**Describe the pull request**

- #### What does your PR fix?
  Update simdjson to v2.0.0.

  Currently, the newest release in upstream is 2.2.0. I am planning to catch up upstream with several PRs:
  - 1.1.0 -> 2.0.0 -> 2.0.1 -> 2.0.2 -> 2.0.3 -> 2.0.4 -> 2.1.0 -> 2.2.0.

  (or maybe within one PR).

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  All, Yes.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes.

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/simdjson/portfile.cmake [hunk 1]
```diff
@@ -1,9 +1,9 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO simdjson/simdjson
-    REF v1.1.0
+    REF v2.0.0
```
Comment (FrankXie05) [inline ports/simdjson/portfile.cmake:4]:
```suggestion
    REF 1075e8609c4afa253162d441437af929c29e31bb #v2.2.0
```

**Review by FrankXie05 — CHANGES_REQUESTED** [review]
We hope that the `REF` attached to the commit id is not just a tag number

**Comment by myd7349** [comment]
> 

Thanks. Done.

**Comment by myd7349** [comment]
> @myd7349 Thanks for your PR, Why not update to version 2.2.0 directly?

Thanks. Done.

File: versions/s-/simdjson.json [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "2bc3bd74a2cb0719ba123f23538e9e974f88c320",
+      "version-semver": "2.2.0",
+      "port-version": 0
+    },
+    {
+      "git-tree": "4fad91e3589f242dbdd1e698b87f78eba9e6338b",
+      "version-semver": "2.0.0",
+      "port-version": 0
+    },
```
Comment (FrankXie05) [inline versions/s-/simdjson.json:12]:
```suggestion
```

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

**Review by FrankXie05 — APPROVED** [review]

**Pull request merged**